### PR TITLE
Add prettier and npm install API

### LINKED ISSUE #446: Init: add API to launch npm install and prettier
In Init context, add API to launch `npm install` and `npm run prettier:format` from CommandRepository

### PR DESCRIPTION
**Pull request by swarajsaaj**
Fix #446

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -6,3 +6,5 @@ package-lock.json
 .mvn
 gradle
 .gradle
+
+src/test/resources/generator/init/TestClass.java
```
Comment (pascalgrimaud) [inline .prettierignore:10]: no need (see my other comment)

File: src/main/java/tech/jhipster/lite/generator/init/infrastructure/primary/rest/InitResource.java [hunk 2]
```diff
@@ -29,4 +29,13 @@ public void init(@RequestBody ProjectDTO projectDTO) {
...
     initApplicationService.init(project);
   }
+
+  @Operation(summary = "Install project")
+  @ApiResponse(responseCode = "500", description = "An error occurred while installing project")
+  @PostMapping("/init/install")
+  public void install(@RequestBody ProjectDTO projectDTO) {
+    Project project = ProjectDTO.toProject(projectDTO);
+    initApplicationService.install(project);
+    initApplicationService.prettify(project);
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/init/infrastructure/primary/rest/InitResource.java:39]: I think it's better to split both:
- 1 for npm install
- 1 for prettify

File: src/main/java/tech/jhipster/lite/generator/project/infrastructure/secondary/executables/Npm.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+public class Npm {
+
+  private Npm() {}
+
+  public static String getExecutableCommand() {
+    if (isWindows()) {
+      return "npm.cmd";
+    } else {
+      return "npm";
+    }
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/project/infrastructure/secondary/executables/Npm.java:14]:
```suggestion
    return isWindows() ? "npm.cmd" : "npm";
```

File: src/test/java/tech/jhipster/lite/TestUtils.java [hunk 4]
```diff
@@ -108,6 +108,12 @@ public static Project tmpProjectWithSpringBootProperties() {
     return project;
   }
 
+  public static Project tmpProjectWithSpringCloudBootstrapProperties() {
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/TestUtils.java:111]: why in this PR ?

File: src/test/java/tech/jhipster/lite/TestUtils.java [hunk 5]
```diff
@@ -165,6 +171,27 @@ public static void copySpringBootProperties(Project project) {
     }
   }
 
+  private static void copySpringCloudBootstrapProperties(Project project) {
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/TestUtils.java:174]: why in this PR ? :-D

File: src/test/java/tech/jhipster/lite/generator/init/application/InitAssertFiles.java [hunk 6]
```diff
@@ -50,4 +55,34 @@ public static void assertFilesInit(Project project) {
...
+      throw new AssertionError(e);
+    }
+  }
+
+  public static void assertPrettifiedFile(Project project) {
+    try {
+      assertEquals(
+        FileUtils.read(getPath(project.getFolder(), getPath(MAIN_JAVA, "TestClass.java"))),
+        """
+        public class TestClass {
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/init/application/InitAssertFiles.java:76]: not sure it's a good idea to test the format here, as it's the responsability of prettier lib
so no need to test the result I think, just if the method has been called

File: src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/executables/NpmTest.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import tech.jhipster.lite.UnitTest;
+import tech.jhipster.lite.common.domain.OSUtils;
+
+@UnitTest
+public class NpmTest {
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/executables/NpmTest.java:16]:
```suggestion
class NpmTest {
```

File: src/test/resources/generator/init/TestClass.java [hunk 8]
```diff
@@ -0,0 +1,2 @@
+public class TestClass {
```
Comment (pascalgrimaud) [inline src/test/resources/generator/init/TestClass.java:1]: no need this file

File: src/test/java/tech/jhipster/lite/generator/init/application/InitAssertFiles.java [hunk 9]
```diff
@@ -1,13 +1,18 @@
 package tech.jhipster.lite.generator.init.application;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static tech.jhipster.lite.TestUtils.assertFileExist;
+import static tech.jhipster.lite.common.domain.FileUtils.getPath;
 import static tech.jhipster.lite.common.domain.FileUtils.getPathOf;
+import static tech.jhipster.lite.generator.project.domain.Constants.MAIN_JAVA;
+import static tech.jhipster.lite.generator.project.domain.Constants.MAIN_RESOURCES;
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/init/application/InitAssertFiles.java:9]: not used

File: src/test/java/tech/jhipster/lite/generator/init/domain/InitDomainServiceTest.java [hunk 10]
```diff
@@ -13,8 +13,10 @@
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 import tech.jhipster.lite.UnitTest;
+import tech.jhipster.lite.generator.project.domain.CommandRepository;
 import tech.jhipster.lite.generator.project.domain.Project;
 import tech.jhipster.lite.generator.project.domain.ProjectRepository;
+import tech.jhipster.lite.generator.project.infrastructure.secondary.CommandLocalRepository;
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/init/domain/InitDomainServiceTest.java:19]: not used

File: src/test/java/tech/jhipster/lite/generator/init/infrastructure/primary/rest/InitResourceIT.java [hunk 11]
```diff
@@ -1,17 +1,24 @@
 package tech.jhipster.lite.generator.init.infrastructure.primary.rest;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/init/infrastructure/primary/rest/InitResourceIT.java:3]: not used

File: src/test/java/tech/jhipster/lite/generator/init/infrastructure/primary/rest/InitResourceIT.java [hunk 12]
```diff
@@ -1,17 +1,24 @@
...
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 import static tech.jhipster.lite.TestUtils.*;
-import static tech.jhipster.lite.generator.init.application.InitAssertFiles.assertFilesInit;
+import static tech.jhipster.lite.common.domain.FileUtils.getPath;
+import static tech.jhipster.lite.common.domain.FileUtils.getPathOf;
+import static tech.jhipster.lite.generator.init.application.InitAssertFiles.*;
+import static tech.jhipster.lite.generator.project.domain.Constants.MAIN_JAVA;
 
+import java.io.IOException;
+import java.nio.file.Files;
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/init/infrastructure/primary/rest/InitResourceIT.java:13]: clean some unused import here

File: src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/executables/NpmTest.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
+package tech.jhipster.lite.generator.project.infrastructure.secondary.executables;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static tech.jhipster.lite.common.domain.OSUtils.isWindows;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/executables/NpmTest.java:9]: clean some import here

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
Thanks a lot for your work here @swarajsaaj !

I'm waiting this PR to be merged, then I'll refactor it, as I'm starting the front part.
It means all of this need to be in a specific context, but no worry, let me do the refactoring part

File: src/test/java/tech/jhipster/lite/TestUtils.java (hunk 5 above)
Comment (swarajsaaj) [reply src/test/java/tech/jhipster/lite/TestUtils.java:174]: Oops. Seems some "undo" commands did the messup between branches :P

File: .prettierignore (hunk 1 above)
Comment (swarajsaaj) [reply .prettierignore]: Agreed. removed.

File: src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/executables/NpmTest.java (hunk 13 above)
Comment (swarajsaaj) [reply src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/executables/NpmTest.java]: well i really need to fix my save actions in new IDE :)

**Review by pascalgrimaud — APPROVED** [review]

**Pull request merged**